src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java

```diff
@@ -0,0 +1,136 @@
+package org.openstreetmap.atlas.checks.validation.points;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.tags.AddressHousenumberTag;
+import org.openstreetmap.atlas.tags.AddressStreetTag;
+import org.openstreetmap.atlas.tags.names.NameTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * Tests for {@link AddressPointMatchCheck}
+ *
+ * @author savannahostrowski
+ */
+
+public class AddressPointMatchCheck extends BaseCheck
+{
+    private static final long serialVersionUID = 1L;
+    public static final String NO_STREET_NAME_POINT_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. The street name should likely "
+            + "be one of {1}. These names were derived from nearby Points.";
+    public static final String NO_STREET_NAME_EDGE_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. The street name should likely "
+            + "be one of {1}. These names were derived from nearby Edges.";
+    public static final String NO_SUGGESTED_NAMES_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. No suggestions names were found.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            NO_STREET_NAME_POINT_INSTRUCTIONS, NO_STREET_NAME_EDGE_INSTRUCTIONS,
+            NO_SUGGESTED_NAMES_INSTRUCTIONS);
+    private static final String ADDRESS_STREET_NUMBER_KEY = AddressHousenumberTag.KEY;
+    private static final String POINT_STREET_NAME_KEY = AddressStreetTag.KEY;
+    private static final String EDGE_STREET_NAME_KEY = NameTag.KEY;
+    private static final double BOUNDS_SIZE_DEFAULT = 150.0;
```

Comment:
This seems a bit high. This would create a rectangle with ~300m long sides, which would cover a relatively big area. I'd suggest running with lower values and checking the results. I'd bet with a value like 25 here you'd be able to generate the same results.

Reply:
In some cases where an address belongs to a campus/hospital/large building it can be quite far from any nearby named roads so I think 25 would be a low value. Something closer to 75 might improve performance but catch more of the large building cases.

Reply:
I did a sensitivity analysis on different values and found that 75m yields the best results as a default value for the bounding box size. Any lower and the number of street recommendations drops significantly while increasing the size beyond 75 yields less and less significant results the larger the box becomes.